### PR TITLE
Aliu/galaxy ci

### PR DESCRIPTION
**Pull request by aliuTT**

### CODE REVIEW

File: tt_metal/impl/device/device_pool.cpp [hunk 1]
```diff
@@ -221,9 +199,41 @@ void DevicePool::add_devices_to_pool(
 }
 
 void DevicePool::init_firmware_on_active_devices() const {
+    bool is_galaxy = tt::Cluster::instance().is_galaxy_cluster();
```
Comment (ubcheema) [inline tt_metal/impl/device/device_pool.cpp:202]: Is this needed? Looks like its not used.

**Review by ttmchiou — APPROVED** [review]
Approving for https://github.com/tenstorrent/tt-metal/actions/runs/9518929905/job/26241123327
for related tg pipeline

File: tt_metal/impl/device/device_pool.cpp [hunk 2]
```diff
@@ -225,13 +222,15 @@ void DevicePool::init_firmware_on_active_devices() const {
...
         this->initialize_device(dev);
         detail::InitDeviceProfiler(dev);
-        for (uint32_t t = 0; t < tunnels_from_mmio.size(); t++) {
-            // Need to create devices from farthest to the closest.
-            for (uint32_t ts = tunnels_from_mmio[t].size() - 1; ts > 0; ts--) {
-                uint32_t mmio_controlled_device_id = tunnels_from_mmio[t][ts];
-                log_debug(tt::LogMetal, "Tunnel {} Device {} Tunnel Stop: {}", t, mmio_controlled_device_id, ts);
-                this->initialize_device(this->devices[mmio_controlled_device_id].get());
-                detail::InitDeviceProfiler(this->devices[mmio_controlled_device_id].get());
+        if (not this->skip_remote_devices) {
```
Comment (ubcheema) [inline tt_metal/impl/device/device_pool.cpp:225]: under what conditions are we going to skip remote devices?

Comment (aliuTT) [reply tt_metal/impl/device/device_pool.cpp:216]: Mostly for CI jobs that run on N300. There are a lot of single chip tests in there, without skipping remote devices, the pipelines take twice as long to finish.

Comment (aliuTT) [reply tt_metal/impl/device/device_pool.cpp:216]: It's only used in `CreateDevice(0, skip_remote=true)`

**Pull request merged**